netclient/functions/mqpublish.go

```diff
@@ -140,7 +142,16 @@ func publish(nodeCfg *config.ClientConfig, dest string, msg []byte, qos byte) er
 			err = token.Error()
 		}
 		if err != nil {
-			return token.Error()
+			if brokerErr := checkBroker(nodeCfg.Server.Server); brokerErr != nil {
+				if brokerErr.Error() == "dns" {
```

Comment:
see above, turn lines 146-152 into private func.

Reply:
moved logging to checkBroker func